### PR TITLE
[API][SIMS-#647]PY_Prefix_ApplicationNumber

### PR DESCRIPTION
**Pull request by guru-aot**
- [x] PY should start with the First four digits of the program year example: 2021XXXX  (2021-2022) 2022XXXX  (2022-2023)
- [x] Create new column in PY for PY prefix
- [x] Review constraint of more than 4 varchar -Not required, API to ensure 10 characters, and 0 fill up to 10 characters, pass PY variable and not 2122

image

### CODE REVIEW

File: sources/packages/api/src/services/application/application.service.ts [hunk 1]
```diff
@@ -117,9 +117,13 @@ export class ApplicationService extends RecordDataModelService<Application> {
...
     if (application.applicationStatus === ApplicationStatus.draft) {
-      // TODO:remove the static program year and add dynamic year, from program year table
+      /**
+       * Generate the application number with respect to the programYearPrefix.
+       * This ensures that respective sequence is created in the sequence_controls table
+       * for specific year 2021,2022 subsequently.
+       */
       application.applicationNumber = await this.generateApplicationNumber(
-        "2122",
+        application.programYear.programYearPrefix,
```
Comment (guru-aot) [inline sources/packages/api/src/services/application/application.service.ts:126]: No need to check if the value programYearPrefix is available or not, and throw an exception, as we already have not null constraint in the database level.

File: sources/packages/api/src/database/entities/program-year.model.ts [hunk 2]
```diff
@@ -52,6 +52,15 @@ export class ProgramYear extends RecordDataModel {
     nullable: false,
   })
   partnerFormName: string;
+  /**
+   * Program Year Prefix  for the
```
Comment (andrewsignori-aot) [inline sources/packages/api/src/database/entities/program-year.model.ts:56]: Please remove the additional white space in the comment.

**Review by andrewsignori-aot — APPROVED** [review]
Looks good 👍

**Review by dheepak-aot — APPROVED** [review]
LGTM

**Review by abschwenker — APPROVED** [review]
lgtm

**Pull request merged**